tests/src/test/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandlerTest.java

```diff
@@ -107,6 +114,42 @@ protected void cleanup() throws Exception {
         super.internalCleanup();
     }
 
+    @Test
+    public void testAutoReadEnableDisAble() {
+        final ByteBuf buffer = Unpooled.copiedBuffer("Test", CharsetUtil.US_ASCII);
+        final EmbeddedChannel channel = new EmbeddedChannel(handler);
+        ByteBuf[] buffers = new ByteBuf[1000000];
+        for (int i = 0; i < buffers.length; i++) {
+            buffers[i] = buffer.duplicate().retain();
+        }
+        try {
+            AtomicBoolean isServiceFinished = new AtomicBoolean(false);
+            AtomicBoolean isChannelWritable = new AtomicBoolean(true);
+            ExecutorService service = Executors.newSingleThreadExecutor();
+
+            service.execute(() -> {
+                int count = 0;
+                while (count < 10) {
+                    boolean isWritable = handler.getCtx().channel().isWritable();
+                    if (!isWritable) {
+                        isChannelWritable.set(false);
+                        break;
+                    }
+                    count += 1;
+                    try {
+                        TimeUnit.MILLISECONDS.sleep(50);
+                    } catch (Exception ignored) {}
+
+                }
+                isServiceFinished.set(true);
```

Comment:
Is `isServiceFinished` meaningless here?

Reply:
yes. "writeOutbound" is a sync operation which won't finish until all data have been sent. I was thinking that main thread would finished quickly before detecting writable of channel in an independent thread.